ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/hooks/CommonHooks.kt

```diff
@@ -127,3 +129,32 @@ public object ReceiveRequestBytes : Hook<(call: ApplicationCall, body: ByteReadC
         }
     }
 }
+
+/**
+ * A hook that is invoked before `Transform` phase.
+ * Useful for some plugins which used for templates as views within application.
+ */
+@InternalAPI
+public class ResponseBeforeTransform<T : Any>(private val clazz: KClass<T>) :
+    Hook<suspend (call: ApplicationCall, body: T) -> Any> {
+    override fun install(
+        pipeline: ApplicationCallPipeline,
+        handler: suspend (call: ApplicationCall, body: T) -> Any
+    ) {
+        val beforeTransform = PipelinePhase("BeforeTransform")
+        pipeline.sendPipeline.insertPhaseBefore(ApplicationSendPipeline.Transform, beforeTransform)
+        pipeline.sendPipeline.intercept(beforeTransform) { body ->
+            if (body.instanceOf(this@ResponseBeforeTransform.clazz)) {
+                @Suppress("UNCHECKED_CAST")
+                subject = handler(call, body as T)
+            }
+        }
+    }
+}
+
+@InternalAPI
+public inline fun <reified T : Any> PluginBuilder<*>.onResponseBeforeTransform(
```

Comment:
IMO this function is redundant. Just use `on(Hook) { ... }` API inside the plugins, especially because it's internal API.